EcoSoapBank/Model/ImpactStats.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  ImpactStats.swift
+//  EcoSoapBank
+//
+//  Created by Shawn Gee on 8/11/20.
+//  Copyright © 2020 Spencer Curtis. All rights reserved.
+//
+
+import UIKit
+
+struct ImpactStats: Decodable {
+    
+    let stats: [Statistic: Int]
+
+    enum Statistic: String, CodingKey, CaseIterable {
+        case soapRecycled
+        case linensRecycled
+        case bottlesRecycled
+        case paperRecycled
+        case peopleServed
+        case womenEmployed
+        
+        var description: String {
+            switch self {
+            case .soapRecycled:
+                return "soap recycled"
+            case .linensRecycled:
+                return "linens recycled"
+            case .bottlesRecycled:
+                return "bottle amenities/nrecycled"
+            case .paperRecycled:
+                return "paper recycled"
+            case .peopleServed:
+                return "people served"
+            case .womenEmployed:
+                return "women employed"
+            }
+        }
+        
+        var image: UIImage {
+            switch self {
+            case .soapRecycled:
+                return UIImage(named: "Bottles")!
+            case .linensRecycled:
+                return UIImage(named: "Bottles")!
+            case .bottlesRecycled:
+                return UIImage(named: "Bottles")!
+            case .paperRecycled:
+                return UIImage(named: "Bottles")!
+            case .peopleServed:
+                return UIImage(named: "Bottles")!
+            case .womenEmployed:
+                return UIImage(named: "Bottles")!
+            }
+        }
+    }
+    
+    func amountString(for stat: Statistic) -> String? {
+        if let amount = stats[stat] {
+            return String(amount)
+        } else {
+            return nil
+        }
+    }
```

Comment:
I could see this being something like `"\(stats[stat] ?? 0) \(stat.description)"` (and then the return type could be non-optional), though I'm not sure if that's what you'd want at the call site.

Reply:
The description actually goes on a separate label. I don't plan on asking for an amount that isn't there, and if I did, it should really be nil. And the optionality doesn't affect anything as we are setting it on a label which has optional text.